src/main/java/testapp/adapter/memory/InMemoryUserRepository.java

```diff
@@ -0,0 +1,38 @@
+package testapp.adapter.memory;
+
+import testapp.core.User;
+import testapp.core.UserAlreadyExistsException;
+import testapp.core.UserRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class InMemoryUserRepository implements UserRepository {
+  private List<User> userList = new ArrayList<>();
+
+  @Override
+  public List<User> findUsers() {
+    return userList;
+  }
+
+  @Override
+  public void register(User user) {
+    for (User each : userList) {
+      if (Objects.equals(each, user)) {
```

Comment:
right now you compare the entire objects, so if two users have the same id, but different name they'll both be registered